### DISCUSSION TITLE
[Support] 未引入的文件被执行

### DISCUSSION THREAD

**Discussion by passby6someone** [midwayjs/midway#3480] [Q&A] [answered]
### Describe the problem（描述问题）

我在db文件夹中放入seed.ts文件用于初始化数据库，但是在运行 ```npm run dev``` 的时候没有被任何文件引入的 ```seed.ts``` 文件被执行了

于是我创建了个同样没有被任何文件引入的的 ```test.ts``` 文件，发现还是被执行了

![image](https://github.com/midwayjs/midway/assets/43069149/918823b6-eecf-46dc-8aae-36e61e12c40c)

![image](https://github.com/midwayjs/midway/assets/43069149/5bd9bf06-b934-47b4-a529-005487024f06)

![image](https://github.com/midwayjs/midway/assets/43069149/4aa36957-fd43-4f7c-8093-2dea1569c179)

![image](https://github.com/midwayjs/midway/assets/43069149/dbdb5fbe-1847-49c1-91f2-81724bb11703)

### Midway Versions（Midway 版本）

    "@midwayjs/bootstrap": "^3.12.0",
    "@midwayjs/core": "^3.12.0",
    "@midwayjs/decorator": "^3.12.0",
    "@midwayjs/info": "^3.12.0",
    "@midwayjs/koa": "^3.12.0",
    "@midwayjs/logger": "^2.14.0",
    "@midwayjs/validate": "^3.12.0",

**Comment by czy88840616** [comment]
正常，框架启动会扫描业务目录做依赖树分析。

**Reply by passby6someone** [reply]
我不太理解，为什么扫描文件做依赖树分析会同时执行一次？

按照我的理解，扫描文件做依赖分析应该是用AST之类的看import了哪些文件吧，必须要执行一次么

**Reply by czy88840616** [reply, marked as answer]
参考https://midwayjs.org/docs/container#%E8%87%AA%E5%8A%A8%E6%89%AB%E6%8F%8F%E7%BB%91%E5%AE%9A
